charms/worker/k8s/src/charm.py

```diff
@@ -601,6 +601,48 @@ def _get_scrape_jobs(self):
             log.exception("Failed to get COS token.")
         return []
 
+    @on_error(
+        ops.WaitingStatus("Sharing Cluster Version"),
+        AssertionError,
+    )
+    def _update_kubernetes_version(self):
+        """Update the unit Kubernetes version in the cluster relation."""
+        relation_name = "cluster"
+        if not (relation := self.model.get_relation(relation_name)):
+            assert False, "Missing cluster integration"  # nosec
+        if version := snap_version("k8s"):
+            relation.data[self.unit]["version"] = version
+
+    @on_error(
+        ops.WaitingStatus("Announcing Kubernetes version"),
+        AssertionError,
+    )
+    def _announce_kubernetes_version(self):
+        """Announce the Kubernetes version to the cluster.
+
+        This method ensures that the Kubernetes version is consistent across the cluster.
+        """
+        peer = self.model.get_relation("cluster")
+        worker = self.model.get_relation("k8s-cluster")
+        if not all([peer, worker]):
+            assert False, "Missing cluster integration"
+
+        version = snap_version("k8s")
+        assert version, "k8s-snap is not installed"  # nosec
+
+        for relation in (peer, worker):
+            units = (unit for unit in relation.units if unit.name != self.unit.name)
+            for unit in units:
+                unit_version = relation.data[unit].get("version")
```

Comment:
maybe we can extract the `version` field name since it's used in multiple places.

Reply:
In this process, we’re managing multiple Kubernetes versions. First, we check our local version (the Kubernetes version installed on the leader unit) and confirm whether the versions reported by all other units in the cluster (both control planes and workers) match. If any unit’s version does not align with the leader's, we block the leader unit. This check will be integrated into the upgrade process in a future PR allowing us to determine whether we are upgrading to a supported version or not